src/app/teacher/apply-tags-button/apply-tags-button.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { Project } from '../../domain/project';
+import { TagService } from '../../../assets/wise5/services/tagService';
+
+@Component({
+  selector: 'apply-tags-button',
+  templateUrl: './apply-tags-button.component.html',
+  styleUrls: ['./apply-tags-button.component.scss']
+})
+export class ApplyTagsButtonComponent implements OnInit {
+  protected tags: any[] = [];
+  @Input() selectedProjects: Project[] = [];
+
+  constructor(private tagService: TagService) {}
+
+  ngOnInit(): void {
+    this.tagService.retrieveUserTags().subscribe((tags) => {
+      for (const tag of tags) {
+        tag.checked = this.doesAnyProjectHaveTag(tag);
+      }
+      this.tags = tags;
+    });
+  }
+
+  private doesAnyProjectHaveTag(tag: any): boolean {
+    for (const project of this.selectedProjects) {
+      if (project.tags.includes(tag.text)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  protected toggleTagOnProjects(tag: any, checked: boolean): void {
```

Comment:
The second parameter indicates whether we're adding or removing the tag, so rename variable to match?
```suggestion
  protected toggleTagOnProjects(tag: Tag, addTag: boolean): void {
```